corehq/util/view_utils.py

```diff
@@ -68,6 +69,7 @@ def inner(request, *args, **kwargs):
         except BadRequest as e:
             return _get_json_exception_response(400, request, e)
         except Exception as e:
+            notify_exception(request, 'JSON exception response: {}'.format(e))
```

Comment:
I totally agree that passing strings around as Unicode until we need to render them is the best approach. ... And I also agree that we don't do that often, but we should definitely do it more.

Maybe we could pass the kludge on to utility functions like [`django.utils.encoding.smart_text` and `smart_bytes`](https://docs.djangoproject.com/en/1.9/ref/unicode/#useful-utility-functions) to keep our own code less kludgy. 

(Do we have a policy of considering all bytestrings to be UTF-8 unless we have a good reason to think they aren't?)